src/components/calculator.tsx

```diff
@@ -100,14 +102,24 @@ export default function Calculator() {
     setScreenValue(factorial(screenValue));
   };
 
-  const handleSin = () => {
-    setScreenValue(sin(unit(Number(screenValue), "deg")).toString());
-  };
-  const handleCos = () => {
-    setScreenValue(cos(unit(Number(screenValue), "deg")).toString());
-  };
-  const handleTan = () => {
-    setScreenValue(tan(unit(Number(screenValue), "deg")).toString());
+  const handleTrig = (value: string) => {
+    switch (value) {
+      case "sin":
+        setScreenValue(sin(unit(Number(screenValue), "deg")).toString());
+        break;
+
+      case "cos":
+        setScreenValue(cos(unit(Number(screenValue), "deg")).toString());
+        break;
+
+      case "tan":
+        setScreenValue(tan(unit(Number(screenValue), "deg")).toString());
+        break;
+
+      case "asin":
+        setScreenValue(asin(unit(Number(screenValue), "deg")).toString());
+        break;
+    }
   };
```

Comment:
NICEEEEE!!!!
I was just about to say this and you implemented it already 🔥 🔥 
GOOD JOB!!